v0/destinations/braze/transform.js

```diff
@@ -374,7 +376,224 @@ function process(event) {
+  let eventsBatch = [];
+  let purchasesBatch = [];
+  let metadataBatch = [];
+
+  for (let index = 0; index < destEvents.length; index += 1) {
+    // take out a single event
+    const ev = destEvents[index];
+    const { message, metadata, destination } = ev;
+
+    // get the JSON body
+    jsonBody = get(message, "body.JSON");
+
+    // get the type
+    endPoint = get(message, "endpoint");
+    type = endPoint && endPoint.includes("track") ? "track" : "identify";
+
+    // if it is a track keep on adding to the existing track list
+    // keep a count of event, attribute, purchases - 75 is the cap
+    if (type === "track") {
+      // keep the trackEndpoint for reuse later
+      if (!trackEndpoint) {
+        trackEndpoint = endPoint;
+      }
+
+      // look for events, attributes, purchases
+      const { events, attributes, purchases } = jsonBody;
+
+      // add only if present
+      if (attributes) {
+        attributesBatch.push(...attributes);
+      }
+
+      if (events) {
+        eventsBatch.push(...events);
+      }
+
+      if (purchases) {
+        purchasesBatch.push(...purchases);
+      }
+
+      // set the counter to max of len(attributesBatch), len(eventsBatch), len(purchasesBatch)
+      counter = Math.max(
+        attributesBatch.length,
+        eventsBatch.length,
+        purchasesBatch.length
+      );
+
+      // keep the original metadata object. needed later to form the batch
+      metadataBatch.push(metadata);
+
+      // put the cap on 75 or look for the last index
+      if (counter >= BRAZE_MAX_REQ_COUNT || index === destEvents.length - 1) {
+        // form a batch and start over
+        // reuse the last message response for the auth and endpoint
+        // -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+        //  TODO: Need to think about reusing the code
+        // -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+        if (
+          attributesBatch.length > 0 ||
+          eventsBatch.length > 0 ||
+          purchasesBatch.length > 0
+        ) {
+          const batchResponse = defaultRequestConfig();
+          batchResponse.headers = message.headers;
+          batchResponse.endpoint = trackEndpoint;
+          batchResponse.body.JSON = {
+            attributes: attributesBatch,
+            events: eventsBatch,
+            purchases: purchasesBatch,
+            partner: BRAZE_PARTNER_NAME
+          };
+          // modify the endpoint to track endpoint
+          batchResponse.endpoint = trackEndpoint;
+          respList.push(
+            formatBatchResponse(batchResponse, metadataBatch, destination)
+          );
+
+          // clear the arrays and reuse
+          attributesBatch = [];
+          eventsBatch = [];
+          purchasesBatch = [];
+          metadataBatch = [];
+        }
+        // -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+      }
+    } else {
+      // identify
```

Comment:
can we use a lodash groupBy and iterate identify and track events separately.
This can improve batching when the event types are interleaved.